File: JavaScript/src/agingStrategies/Aged.js
// Increases quality of an item over time
export default class Aged {
  static tick(item) {
    item.daysRemaining -= 1;

    item.quality += 1;
    if (item.daysRemaining <= 0) {
      item.quality += 1;
    }

    item.quality = Math.min(item.quality, this.#maximumQuality());
  }

  static #maximumQuality() {
    return 50;
  }
}

File: Ruby/lib/aging_strategies/aged.rb
# frozen_string_literal: true

module AgingStrategies
  # Increases quality of an item over time
  class Aged
    def self.tick(item)
      item.days_remaining -= 1

      item.quality += 1
      item.quality += 1 if item.days_remaining <= 0

      item.quality = [item.quality, maximum_quality].min
    end

    private_class_method def self.maximum_quality
      50
    end
  end
end
